app/controllers/products_controller.rb

```diff
@@ -32,7 +37,7 @@ def update
 
   def destroy
     @product.destroy
```

Comment:
let's destroy products in one request, not one object per request, so update finder to find products by array of ids, and then destroy them all in destroy method
https://stackoverflow.com/questions/14434071/how-to-destroy-multiple-objects-simultaneously-in-rails-3